### DISCUSSION TITLE
apk with ndk and language like Vlang and Golang ...

### DISCUSSION THREAD

**Discussion by MatejMagat305** [termux/termux-app#3356] [General]
well I was trying new language Vlang and I was found way how make ndk apk with this lang ([video](https://www.youtube.com/watch?v=7aUh39w_-2Q)), but when I want to make this with Golang it write error:
```
go build -ldflags -w -trimpath -buildmode=c-shared -o /data/data/com.termux/files/usr/tmp/fyne-work-902498925/lib/arm64-v8a/libe.so basic failed: exit status 1
# runtime/cgo
cgo: exec /data/data/com.termux/files/home/android-ndk-r23c/toolchains/llvm/prebuilt/linux-aarch64/bin/aarch64-linux-android31-clang: fork/exec /data/data/com.termux/files/home/android-ndk-r23c/toolchains/llvm/prebuilt/linux-aarch64/bin/aarch64-linux-android31-clang: no such file or directory
```
I has writed on Golang issues ([isue](https://github.com/golang/go/issues/58587)), but they claim that it is error of termux, of course I was running this after set PATHs from video and use fyne.io where I was trying make change ([PR](https://github.com/fyne-io/fyne/pull/3649/files))... , 
what do you think is it really bug of termux or it is bug of Golang?

**Comment by sylirre** [comment]
Few questions:

* Do you really have NDK in Termux installed?
* Where you got this NDK, if you have it installed? Are you sure it is built for Termux specifically or just a generic build for Linux?

Reminding that Termux is quite specialized environment which is not compatible with standard Linux systems: https://wiki.termux.com/wiki/Differences_from_Linux

NDK from Google will not work in any way under Termux, unless your device is x86_64 and you run a chroot environment.

**Comment by MatejMagat305** [comment]
the NDK is/was instaled by my script https://github.com/MatejMagat305/vab-termux
from https://github.com/Lzhiyong/termux-ndk, the proof is video, so yes it was especially for termux (arm64)

**Reply by sylirre** [reply]
Well, that's better. But even it has some issues. Perpahs you are facing "no such file or directory" because of shebang:

![Screenshot_20230424-095212](https://user-images.githubusercontent.com/107305601/233920696-d19c7fbb-c7ad-4165-9583-3e2f0572ac2c.png)

The clang wrapper scripts use `#!/bin/bash` but path `/bin/bash` does not exist under Termux. These scripts may still work if `termux-exec` works on your device and LD_PRELOAD variable is not being unset (e.g. by Golang). Use `termux-fix-shebang` to fix wrappers:

![Screenshot_20230424-095517](https://user-images.githubusercontent.com/107305601/233921321-84e316e1-52fd-4738-b1bd-9f0583a52e0e.png)

**Reply by MatejMagat305** [reply]
au, thank
